addons/stock/models/stock_move.py

```diff
@@ -654,8 +654,15 @@ def action_assign_serial(self):
     def action_product_forecast_report(self):
         self.ensure_one()
         action = self.product_id.action_product_forecast_report()
+        action['context'] = {
+            'active_id': self.product_id.id,
+            'active_model': 'product.product',
+            'order_name': self.origin,
+            'picking_name': self.picking_id.name,
```

Comment:
I think it's useless to have a separate key for order and picking, as you get `order_name` in priority.
```suggestion
            'order_name': self.origin or self.picking_id.name,
```
This way, it's easier when you get back the order name in `_get_report_data`.

Also, `origin` isn't always an order name (editable field, could be anything, sometime multiple orders names), maybe it's better to use the fields `purchase_id` and/or `sale_id` instead.